### PR TITLE
A0-3432: E2E tests run now with RPC node as bootnode.

### PR DESCRIPTION
**Pull request by Marcin-Radecki**
# Description

Following up https://github.com/Cardinal-Cryptography/aleph-node/commit/21ead16619c24e8e5197e8de1c368905a604141a, we introduce RPC Node to e2e docker-compose framework. This is main, non breaking change to anybody apart from e2e package crate system (CI). Other than that, below fixes have been made:
* `run_consensus.sh` now runs only amount of nodes user passes, now all nodes (8) in the common docker compose file, to not waste CPU/memory
* RPC_PORT now it's 9944 - this was last place it was non-unified,
* setting minimum validator count is not required, meaning that from e2e test perspective it does not matter if we pass 7 as currently we do or it is default 4 value - tests passes anyway. Hence, we removed that feature
* `run_consensus.sh` now can be run from any path,
* common docker-compose now supports up to 9 validators
* stash now equals to controller in chain_spec generation. This code was dead anyway as substrate 1.0.0 skips controller anyway: https://github.com/Cardinal-Cryptography/substrate/blob/aleph-v1.0.0/frame/staking/src/pallet/mod.rs#L618
* removed not used `RANDOMIZED` feature in e2e tests
* updated sync and latency tests to point to 9944 port

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Testing:

- nightly e2e logic tests pass: https://github.com/Cardinal-Cryptography/aleph-node/actions/runs/6783539788
- normal session e2e tests: pass https://github.com/Cardinal-Cryptography/aleph-node/actions/runs/6795064373

### CODE REVIEW

File: .github/scripts/run_consensus.sh [hunk 1]
```diff
@@ -60,61 +51,91 @@ while [[ $# -gt 0 ]]; do
...
-
-  echo "Generating ${authorities_count} authorities accounts ids..." >&2
-  declare -a account_ids
-  for node_index in $(seq 0 $((authorities_count - 1))); do
-    echo "Generating authority ${node_index} from key //${node_index}" >&2
-    account_ids+=($(docker run --rm -v $(pwd)/docker/data:/data --entrypoint "/bin/sh" "${NODE_IMAGE}" -c "aleph-node key inspect //$node_index | grep \"SS58 Address:\" | awk \"{print \\\$3;}\""))
+  echo "Generating one RPC node and ${validators_count} validators accounts ids..." >&2
+  local account_ids=()
+  for node_index in $(seq 0 "${validators_count}"); do
+    echo "Generating account if ${node_index} from key //${node_index}" >&2
```
Comment (fixxxedpoint) [inline .github/scripts/run_consensus.sh:62]: `if`?

File: e2e-tests/src/config.rs [hunk 2]
```diff
@@ -134,11 +134,11 @@ pub struct Config {
     /// WS endpoint address of the node to connect to
     pub node: String,
 
-    /// Number of //0, //1, ... validators to run e2e tests on
+    /// Number validators to run e2e tests on
```
Comment (fixxxedpoint) [inline e2e-tests/src/config.rs:137]: missing `of`

File: scripts/synthetic-network/README.md [hunk 3]
```diff
@@ -21,33 +21,35 @@ Additionally, this folder contains an example .js script introducing API of the
...
-# copy created binary to e2e-tests/target/release/, built test binary is in the last line of
-# the above command, e.g.
-# cp target/release/deps/aleph_e2e_client-44dc7cbed6112daa target/release/aleph-e2e-client
-docker build --tag aleph-e2e-client:latest -f Dockerfile .
-cd ..
-.github/scripts/run_e2e_test.sh -t high_out_latency_for_all -m 5
+# run tests for the block sync component
+aleph-node$ VALIDATOR_COUNT=7 NETWORK="synthetic-network" NODE_URL="ws://Node0:9944" ./.github/scripts/run_e2e_test.sh -t test::sync -n 7
+# another example
+.github/scripts/run_e2e_test.sh -t high_out_latency_for_all
```
Comment (fixxxedpoint) [inline scripts/synthetic-network/README.md:46]: missing `aleph-node$`

**Review by fixxxedpoint — APPROVED** [review]
lgtm

**Review by maciejzelaszczyk — APPROVED** [review]

File: scripts/synthetic-network/README.md (hunk 3 above)
Comment (Marcin-Radecki) [reply scripts/synthetic-network/README.md]: Done

File: e2e-tests/src/config.rs (hunk 2 above)
Comment (Marcin-Radecki) [reply e2e-tests/src/config.rs]: Done

File: .github/scripts/run_consensus.sh (hunk 1 above)
Comment (Marcin-Radecki) [reply .github/scripts/run_consensus.sh]: Done

File: .github/scripts/run_consensus.sh [hunk 4]
```diff
@@ -1,38 +1,34 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-set -euo pipefail
+# This script bootstraps and runs aleph-node consensus for e2e tests.
+# It is not intended to by used by end user, use [run_nodes.sh](../../scripts/run_nodes.sh) for
```
[the remark this answers is not in the document]
Comment (Marcin-Radecki) [reply .github/scripts/run_consensus.sh]: Done

**Pull request merged**